src/Microsoft.Framework.Runtime.Roslyn.Interfaces/IProjectContext.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Runtime.Versioning;
+
+namespace Microsoft.Framework.Runtime.Roslyn
+{
+    /// <summary>
+    /// Stores information about the project
+    /// </summary>
+    public interface IProjectContext
```

Comment:
If we add this, could you ensure `IApplicationEnvironment` is removed from precompilation? Also, this might need the equivalent of `IApplicationEnvironment.Configuration`. Mvc needs that value for compilation options parsing